### ISSUE TITLE
Error: EMFILE: too many open files

### ISSUE THREAD

**Issue by ariel-frischer** [serverless/serverless#9073]
`serverless deploy` is running into global config errors and Error: EMFILE: too many open files. I've tried uninstalling and reinstalling global serverless command as well as restarting my computer. It used to deploy until I tried to upgrade my global serverless with `npm i -g serverless/serverless`.

<details>
<summary><code>serverless.yml</code></summary>

```yaml
service: aws-node-puppeteer
app: initiate-web
org: arielf

provider:
  name: aws
  runtime: nodejs12.x
  region: us-west-1

plugins:
  - serverless-offline
  - serverless-plugin-chrome

package:
  exclude:
    - node_modules/puppeteer/.local-chromium/**
    - layer/**
    - tmp/**
    - .vscode/*
    - yarn.*
    - .eslintrc.json
    - .prettierrc

functions:
  urlToVideo:
    handler: handler.urlToVideo
    memorySize: 3000
    timeout: 30
    environment:
      FFMPEG_PATH: /opt/ffmpeg/ffmpeg
      FFPROBE_PATH: /opt/ffmpeg/ffprobe
    events:
      - http:
          path: /
          method: get
```

</details>

<details>
<summary><b><code>sls deploy</code> output</b></summary>

```
Serverless: Deprecation warning: Resolution of lambda version hashes was improved with better algorithm, which will be used in next major release.
            Switch to it now by setting "provider.lambdaHashingVersion" to "20201221"
            More Info: https://www.serverless.com/framework/docs/deprecations/#LAMBDA_HASHING_VERSION_V2
Serverless: Deprecation warning: Starting with next major version, API Gateway naming will be changed from "{stage}-{service}" to "{service}-{stage}".
            Set "provider.apiGateway.shouldStartNameWithService" to "true" to adapt to the new behavior now.
            More Info: https://www.serverless.com/framework/docs/deprecations/#AWS_API_GATEWAY_NAME_STARTING_WITH_SERVICE
Serverless: Injecting Headless Chrome...
Serverless: Packaging service...
Serverless: Excluding development dependencies...
Serverless: User Configuration warning: Cannot resolve global config file: C:\Users\Ariel\.serverlessrc 
Error: EMFILE: too many open files, open 'C:\Users\Ariel\.serverlessrc'
Serverless: Your previous global config was renamed to C:\Users\Ariel\.serverlessrc.bak for debugging. Default global config will be recreated under C:\Users\Ariel\.serverlessrc.

  Serverless Error ---------------------------------------

  Cannot read file node_modules\moment\src\locale\th.js due to: EMFILE: too many open files, open 'C:\Users\Ariel\sls-node-puppeteer\.build\node_modules\moment\src\locale\th.js'

  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Issues:        forum.serverless.com

  Your Environment Information ---------------------------
     Operating System:          win32
     Node Version:              12.19.0
     Framework Version:         2.28.6
     Plugin Version:            4.4.3
     SDK Version:               2.3.2
     Components Version:        3.7.2```
```
</details>

<b>Installed version</b>

```
Framework Core: 2.28.6
Plugin: 4.4.3
SDK: 2.3.2
Components: 3.7.2
```
<details>

<summary><b><code>package.json</code> dependencies</b></summary>

```
  "dependencies": {
    "chrome-launcher": "^0.13.4",
    "puppeteer": "^2.1.0",
    "serverless-offline": "^6.8.0",
    "serverless-plugin-chrome": "^1.0.0-55.3",
    "timesnap-core": "^0.2.0"
  },
  "devDependencies": {
    "eslint": "^7.20.0",
    "eslint-config-prettier": "^8.1.0",
    "eslint-plugin-prettier": "^3.3.1",
    "prettier": "^2.2.1",
    "prettier-eslint": "^12.0.0"
  }
```
</details>

**Comment by garethmcc** [comment]
@ariel-frischer Is this from running locally?

**Comment by ariel-frischer** [comment]
@garethmcc No, it works locally, the command `sls deploy` is the issue but I feel maybe I need to reset global configuration somehow?

**Comment by garethmcc** [comment]
What I meant was this was from running on your local machine and not an external CI/CD system for example?

**Comment by ariel-frischer** [comment]
Oh, yea this is on my local machine.

**Comment by willjeuniaux** [comment]
I have the same issue happening since 02/03. My service is being built on an external CI/CD. I've identified the error started happening since serverless package version 2.28.3. To work around the problem I am pulling serverless@2.28.0 instead of the latest one. Hope this helps.

**Comment by Ksmike** [comment]
I'm getting the same issue with 2.28.6 via CircleCi, I haven't tested any of the other versions. I'm just about to do a test deploy on 2.28.2 to see if that solves the issue. Basically just ignoring all March 2nd deployments. Will report back back later.

Circle Conf ---
     Operating System:          linux
     Node Version:              12.13.1
     Framework Version:         2.28.6
     Plugin Version:            4.4.3
     SDK Version:               2.3.2
     Components Version:        3.7.2

**Comment by medikoo** [comment]
@ariel-frischer @ Ksmike  if possible please investigate with which Framework version it started to happen.

I assume it could be 2.28.1 when it started to happen (2.28.1 bad, 2.28.0 ok), but do not have service with large number of files to be processed  handy to confirm.

**Comment by willjeuniaux** [comment]
> @ariel-frischer @Ksmike if possible please investigate with which Framework version it started to happen.
> 
> I assume it could be 2.28.1 when it started to happen (2.28.1 bad, 2.28.0 ok), but do not have service with large number of files to be processed handy to confirm.

I confirm it fails on 2.28.1 but works on 2.28.0.

```
yarn global add serverless@2.28.1
success Installed "serverless@2.28.1" with binaries:
      - serverless
      - sls
Done in 37.80s.
Serverless: Excluding development dependencies...
Serverless: User Configuration warning: Cannot resolve global config file: C:\Users\appveyor\.serverlessrc 
Error: EMFILE: too many open files, open 'C:\Users\appveyor\.serverlessrc'
Serverless: Your previous global config was renamed to C:\Users\appveyor\.serverlessrc.bak for debugging. Default global config will be recreated under C:\Users\appveyor\.serverlessrc.
 
  Serverless Error ---------------------------------------
 
  Cannot read file node_modules\assert-plus\README.md due to: EMFILE: too many open files, open 'C:\projects\node_modules\assert-plus\README.md'
```

**Comment by Ksmike** [comment]
> > @ariel-frischer @Ksmike if possible please investigate with which Framework version it started to happen.
> > I assume it could be 2.28.1 when it started to happen (2.28.1 bad, 2.28.0 ok), but do not have service with large number of files to be processed handy to confirm.
> 
> I confirm it fails on 2.28.1 but works on 2.28.0.
> 
> ```
> yarn global add serverless@2.28.1
> success Installed "serverless@2.28.1" with binaries:
>       - serverless
>       - sls
> Done in 37.80s.
> Serverless: Excluding development dependencies...
> Serverless: User Configuration warning: Cannot resolve global config file: C:\Users\appveyor\.serverlessrc 
> Error: EMFILE: too many open files, open 'C:\Users\appveyor\.serverlessrc'
> Serverless: Your previous global config was renamed to C:\Users\appveyor\.serverlessrc.bak for debugging. Default global config will be recreated under C:\Users\appveyor\.serverlessrc.
>  
>   Serverless Error ---------------------------------------
>  
>   Cannot read file node_modules\assert-plus\README.md due to: EMFILE: too many open files, open 'C:\projects\node_modules\assert-plus\README.md'
> ```

I can also confirm it breaking on 2.28.1, but working on 2.28.0.

**Comment by medikoo** [comment]
@willjeuniaux  @Ksmike  great thanks for confirmation. We've coined the issue, it'll be fixed with https://github.com/serverless/serverless/pull/9074 and published shortly

**Comment by Ksmike** [comment]
> @willjeuniaux @Ksmike great thanks for confirmation. We've coined the issue, it'll be fixed with #9074 and published shortly

@medikoo Thanks for the help, I'll let Support know this is being handled! Looking forward to the patch!

**Comment by pgrzesik** [comment]
New version `2.28.7` that should fix that issue has been just released
